### DISCUSSION TITLE
Account in construction

### DISCUSSION THREAD

**Discussion by callmeraymon** [nilsteampassnet/TeamPass#3610] [Q&A]
I just fired up a version [3.0.0.23](https://3.0.0.23/) docker-compose image on my Synology. It's been running for a bit now and I still have "Account in construction" with an hourglass. On hover it reads "Be informed that your password keys are currently in process of being created. Please wait a few minutes more."

I'm not sure what is still under construction. I'm able to create users and passwords. Is anyone else seeing this?

**Comment by nilsteampassnet** [comment]
@callmeraymon 
You are right. After investigation and fixing a set of test scenarios, I identified this too.
I have provided release 3.0.1.

**Reply by azertyuiopssxa** [reply]
Hello, 

I have the same problem with the last version 3.0.1

Thanks

**Reply by jussara-ti** [reply]
Hi, 

it's my first time upgrading Teampass, I don't use Docker but I'm having issues in 3.0.1 upgrade too. I need to upgrade version by version, like upgrade to v3.0.0.23 before go to v3.0.1 or can I go straight to the latest version? Currently, my version is v3.0.0.22

**Reply by azertyuiopssxa** [reply]
Hi,

I have tested the new release (3.0.2) but the problem is always present. 
When I create a user, the different task of creario processus was not executed:  
![image](https://user-images.githubusercontent.com/125877378/228455646-54b86622-9a3a-424f-bc0e-7272f4ee5797.png)
![image](https://user-images.githubusercontent.com/125877378/228455694-179b44d5-3626-48d7-b564-c4d7b6eda9c4.png)

So, I have this message in the user space:
![image](https://user-images.githubusercontent.com/125877378/228456308-a8ac49f7-5a77-4b6c-80b4-b72b9968803f.png)


Thk

**Comment by nilsteampassnet** [comment]
@callmeraymon What do you have in settings tab?

**Reply by callmeraymon** [reply]
Not sure what you mean, there is a lot of stuff in the settings tab. But at the moment, I'm having trouble pulling a new image. Console still reads version 3.0.0.23

**Comment by callmeraymon** [comment]
I rebuild the docker container on ububtu. version 3.0.3 same "account in construction" banner. I did notice that my user, OTV, and API accounts have an hour glass
![hourglass](https://user-images.githubusercontent.com/20956513/229404130-78e95013-bd05-4b08-89c2-42a48375d9d7.PNG)

**Comment by nilsteampassnet** [comment]
The problem comes from the cron job that doesn't seem to be running as it should.
You need to ensure that the job is defined in crontab for the server user used (example www-data), and ensure that this user has rights on the teampass files

**Comment by calebgcooper** [comment]
thanks @nilsteampassnet  seeing this issue as well, what script creates the cron job and what script should the cronjob run?

https://documentation.teampass.net/#/manage/tasks mentions:
sources/scheduler.php

But this is giving me a back trace:
```
teampass:/var/www/html# sudo --preserve-env --user=nginx /usr/local/bin/php /var/www/html/sources/scheduler.php 
PHP Deprecated:  Optional parameter $encryptionKey declared before required parameter $nbItemsToTreat is implicitly treated as a required parameter in /var/www/html/sources/main.functions.php on line 3926
PHP Stack trace:
PHP   1. {main}() /var/www/html/sources/scheduler.php:0

Deprecated: Optional parameter $encryptionKey declared before required parameter $nbItemsToTreat is implicitly treated as a required parameter in /var/www/html/sources/main.functions.php on line 3926

Call Stack:
    0.0003     408424   1. {main}() /var/www/html/sources/scheduler.php:0

PHP Deprecated:  Optional parameter $deleteExistingKeys declared before required parameter $nbItemsToTreat is implicitly treated as a required parameter in /var/www/html/sources/main.functions.php on line 3926
PHP Stack trace:
PHP   1. {main}() /var/www/html/sources/scheduler.php:0

Deprecated: Optional parameter $deleteExistingKeys declared before required parameter $nbItemsToTreat is implicitly treated as a required parameter in /var/www/html/sources/main.functions.php on line 3926

Call Stack:
    0.0003     408424   1. {main}() /var/www/html/sources/scheduler.php:0

PHP Deprecated:  Optional parameter $sendEmailToUser declared before required parameter $nbItemsToTreat is implicitly treated as a required parameter in /var/www/html/sources/main.functions.php on line 3926
PHP Stack trace:
PHP   1. {main}() /var/www/html/sources/scheduler.php:0

Deprecated: Optional parameter $sendEmailToUser declared before required parameter $nbItemsToTreat is implicitly treated as a required parameter in /var/www/html/sources/main.functions.php on line 3926

Call Stack:
    0.0003     408424   1. {main}() /var/www/html/sources/scheduler.php:0

PHP Deprecated:  Optional parameter $encryptWithUserPassword declared before required parameter $nbItemsToTreat is implicitly treated as a required parameter in /var/www/html/sources/main.functions.php on line 3926
PHP Stack trace:
PHP   1. {main}() /var/www/html/sources/scheduler.php:0

Deprecated: Optional parameter $encryptWithUserPassword declared before required parameter $nbItemsToTreat is implicitly treated as a required parameter in /var/www/html/sources/main.functions.php on line 3926

Call Stack:
    0.0003     408424   1. {main}() /var/www/html/sources/scheduler.php:0

PHP Deprecated:  Optional parameter $generate_user_new_password declared before required parameter $nbItemsToTreat is implicitly treated as a required parameter in /var/www/html/sources/main.functions.php on line 3926
PHP Stack trace:
PHP   1. {main}() /var/www/html/sources/scheduler.php:0

Deprecated: Optional parameter $generate_user_new_password declared before required parameter $nbItemsToTreat is implicitly treated as a required parameter in /var/www/html/sources/main.functions.php on line 3926

Call Stack:
    0.0003     408424   1. {main}() /var/www/html/sources/scheduler.php:0

PHP Warning:  Cannot modify header information - headers already sent by (output started at /var/www/html/sources/main.functions.php:3926) in /var/www/html/sources/main.functions.php on line 40
PHP Stack trace:
PHP   1. {main}() /var/www/html/sources/scheduler.php:0
PHP   2. require_once() /var/www/html/sources/scheduler.php:38
PHP   3. header($header = 'Content-type: text/html; charset=utf-8') /var/www/html/sources/main.functions.php:40

Warning: Cannot modify header information - headers already sent by (output started at /var/www/html/sources/main.functions.php:3926) in /var/www/html/sources/main.functions.php on line 40

Call Stack:
    0.0003     408424   1. {main}() /var/www/html/sources/scheduler.php:0
    0.0066    1001272   2. require_once('/var/www/html/sources/main.functions.php') /var/www/html/sources/scheduler.php:38
    0.0066    1001272   3. header($header = 'Content-type: text/html; charset=utf-8') /var/www/html/sources/main.functions.php:40

PHP Warning:  Cannot modify header information - headers already sent by (output started at /var/www/html/sources/main.functions.php:3926) in /var/www/html/sources/main.functions.php on line 41
PHP Stack trace:
PHP   1. {main}() /var/www/html/sources/scheduler.php:0
PHP   2. require_once() /var/www/html/sources/scheduler.php:38
PHP   3. header($header = 'Cache-Control: no-cache, must-revalidate') /var/www/html/sources/main.functions.php:41

Warning: Cannot modify header information - headers already sent by (output started at /var/www/html/sources/main.functions.php:3926) in /var/www/html/sources/main.functions.php on line 41

Call Stack:
    0.0003     408424   1. {main}() /var/www/html/sources/scheduler.php:0
    0.0066    1001272   2. require_once('/var/www/html/sources/main.functions.php') /var/www/html/sources/scheduler.php:38
    0.0068    1001240   3. header($header = 'Cache-Control: no-cache, must-revalidate') /var/www/html/sources/main.functions.php:41

teampass:/var/www/html# 
```

**Reply by calebgcooper** [reply]
I failed to note those are only warnings. So actually no issue after creating cron jobs.

**Comment by Mwendlin** [comment]
Hi,
 i have the same problem by running teampass on a windowss server.
is there any way to get the whole thing running under windows at this time?
The manual itself is unfortunately silent on the subject.

![cronjob](https://github.com/nilsteampassnet/TeamPass/assets/125570883/b279176e-9831-48e2-9288-5d633687b1c2)

**Reply by Mwendlin** [reply]
i found this https://github.com/nilsteampassnet/TeamPass/issues/3778
but have no idea which kind of script i need to write

**Comment by Arodriguesmg** [comment]
Hello,
 
I'm experiencing the same problem and I see that the task manager issue is not resolved. I did some tests on 3.0.7 and it also doesn't run, does anyone know how to perform these tasks manually?

![image](https://github.com/nilsteampassnet/TeamPass/assets/128071906/2f209951-3a96-41cf-a89e-986e2f67871b)
![image](https://github.com/nilsteampassnet/TeamPass/assets/128071906/6371f917-7e6d-48d8-aceb-17d63d1528bb)
![image](https://github.com/nilsteampassnet/TeamPass/assets/128071906/1b7a4c7d-d86c-47a6-84fb-246ffd6aabf1)


Has anyone managed to resolve these issues?

**Comment by is7** [comment]
Hi, 

Possible reason is that on clean install table `processes` has no column `started_at`, but in scripts/background_tasks___user_keys_creation.php:

```php
        DB::update(
            prefixTable('processes'),
            array(
                'started_at' => time(),
            ),
            'increment_id = %i',
            $process_to_perform['increment_id']
        );
```

**Comment by grootsys24** [comment]
I have the same problem. Setup is a fresh Debian 12 server with PHP 8.2 - Taks will not work. User creation does not work.

Cron:
php -f /var/www/html/teampass/sources/scheduler.php

server info:

Debian GNU/Linux 12 (bookworm)

PHP version: 8.2.7
Memory limit: 1024M
Memory usage: 833.66 KB
Maximum time execution: 60
Maximum file size upload: 100M

Server version: 10.11.3-MariaDB-1
Database size: 2.47MB

**Comment by Arodriguesmg** [comment]
Hello guys,
I forgot to send my settings on the server, below:

![image](https://github.com/nilsteampassnet/TeamPass/assets/128071906/ac0bcf74-0dfc-43f4-b5c5-e7a3b63e56b3)

**Reply by Arash9703** [reply]
I have the same problem here ! Anyone got any solution

**Reply by nilsteampassnet** [reply]
@Arash9703 Can you provide the version used? any error in apache logs?

**Reply by Arash9703** [reply]
![image](https://github.com/nilsteampassnet/TeamPass/assets/143269985/209cf90d-c3ea-45ab-836c-edb6ce3b9a71)


![image](https://github.com/nilsteampassnet/TeamPass/assets/143269985/e4239b2b-cd71-4b34-a6b0-4d6b28b0e68d)

**Reply by Arash9703** [reply]
For information, since i started the cronjob i can list my AD Roles but not users , kind of  bizarre :/ 
![image](https://github.com/nilsteampassnet/TeamPass/assets/143269985/176b0593-af3a-4540-b2de-ae80dd560184)
![image](https://github.com/nilsteampassnet/TeamPass/assets/143269985/513def32-cdfa-4730-b84b-a38e31f7d9a4)
![image](https://github.com/nilsteampassnet/TeamPass/assets/143269985/19d6e16d-6afd-4290-9c6c-4b20702d563b)

getting the "in progress" notification and no users loaded even the local user the i created is not activated yet 
![image](https://github.com/nilsteampassnet/TeamPass/assets/143269985/d03640bc-3bcd-4030-a99a-283714cff93b)

**Comment by Arash9703** [comment]
> @Arash9703 Can you provide the version used? any error in apache logs?
The Apache error log is long, and I didn't want to bother you with that. Instead, I shared the error I encountered when running the script for cronjob

**Reply by nilsteampassnet** [reply]
@Arash9703 
Thank you for the error.
While looking a round Google answers with this error.
Most answers refer to mysqli server installation or setup issue.
See https://stackoverflow.com/a/64430573 which is quite interesting.
Some others propose to reinstall MYSQLI.
Also one solution at TP level, could be to "comment" this instruction.
Open file `includes/config/settings.php`
Find
```
define("DB_CONNECT_OPTIONS", array(
    MYSQLI_OPT_CONNECT_TIMEOUT => 10
));
```
and replace by
```
/*define("DB_CONNECT_OPTIONS", array(
    MYSQLI_OPT_CONNECT_TIMEOUT => 10
));*/
```

Please keep us informed

**Reply by Arash9703** [reply]
Thanks @nilsteampassnet , solved the problem . 

Now when i connect to teampass my users teampass session (vault) is not stable (doesn't stop jumping " reloading every second " ). 

![2023-10-17 14_32_18-érreur](https://github.com/nilsteampassnet/TeamPass/assets/143269985/24025e73-2558-4627-accf-536c360eb5b0)

As i connect my self as the local admin user all is well . 
![2023-10-17 14_52_12-Parsec](https://github.com/nilsteampassnet/TeamPass/assets/143269985/0c3bba22-9c29-44ec-af71-3898c1b45239)


I dont' know if i got to make any changes for the users session page refresh via my local admin consol ! 

thanks

**Comment by kikimora52** [comment]
I am having this issue when creating new account sync'd from FreeIPA. I found the below error coming from the docker logs

`Fatal error: Uncaught InvalidArgumentException: Invalid CRON field value */0 at position 0 in /var/www/html/includes/libraries/Cron/CronExpression.php on line 231`

I have not troubleshot further. I made a comment to the offending line, reran scheduler.php and it finished creating the account. This is not a suggested fix, but a work around. I needed something working quickly. I like this product so far and I hope it can work for us.
